### ISSUE TITLE
Implement L.Marker.setStyle()

### ISSUE THREAD

**Issue by IvanSanchez** [Leaflet/Leaflet#3788]
It would be nice if Markers would have a `setStyle` method compatible with the GeoJSON `style` function option.

The use case is a GeoJSON layer with point features which became `L.Marker`s, then calling `L.GeoJSON.setStyle()` with a callback that returns marker options instead of path options.

**Comment by AndreaAstronaut** [comment]
I agree, such a feature would be very useful. I was just investigating if it is possible to do so. I am currently looking for a way to 'highlight' a leaflet marker on mouse-over, for example showing a circular background around it. I wanted to do this using css, so setStyle would be great to have here.

**Comment by IvanSanchez** [comment]
@AndreaAstronaut That can be done with `L.Marker.setIcon()`, and is unrelated to this issue's use case.
